types/node-fetch/index.d.ts

```diff
@@ -5,211 +5,215 @@ import { RequestOptions } from "http";
+    
+    interface BlobOptions {
+        type?: string | undefined;
+        endings?: "transparent" | "native" | undefined;
+    }
+    
+    export class Blob {
+        constructor(blobParts?: BlobPart[], options?: BlobOptions);
+        readonly type: string;
+        readonly size: number;
+        slice(start?: number, end?: number): Blob;
+        text(): Promise<string>;
+    }
+    
+    export class Body {
+        constructor(body?: any, opts?: { size?: number | undefined; timeout?: number | undefined });
+        arrayBuffer(): Promise<ArrayBuffer>;
+        blob(): Promise<Blob>;
+        body: NodeJS.ReadableStream;
+        bodyUsed: boolean;
+        buffer(): Promise<Buffer>;
+        json(): Promise<any>;
+        size: number;
+        text(): Promise<string>;
+        textConverted(): Promise<string>;
+        timeout: number;
+    }
+    
+    interface SystemError extends Error {
+        code?: string | undefined;
+    }
+    
+    export class AbortError extends Error {
+        readonly name: "AbortError";
+        constructor(message: string);
+        readonly type: "aborted";
+    }
+    
+    export class FetchError extends Error {
+        name: "FetchError";
+        constructor(message: string, type: string, systemError?: SystemError);
+        type: string;
+        code?: string | undefined;
+        errno?: string | undefined;
+    }
+    
+    export class Response extends Body {
+        constructor(body?: BodyInit, init?: ResponseInit);
+        static error(): Response;
+        static redirect(url: string, status: number): Response;
+        clone(): Response;
+        headers: Headers;
+        ok: boolean;
+        redirected: boolean;
+        status: number;
+        statusText: string;
+        type: ResponseType;
+        url: string;
+    }
+    
+    export type ResponseType =
+        | "basic"
+        | "cors"
+        | "default"
+        | "error"
+        | "opaque"
+        | "opaqueredirect";
+    
+    export interface ResponseInit {
+        headers?: HeadersInit | undefined;
+        size?: number | undefined;
+        status?: number | undefined;
+        statusText?: string | undefined;
+        timeout?: number | undefined;
+        url?: string | undefined;
+        counter?: number | undefined;
+    }
+    
+    interface URLLike {
+        href: string;
+    }
+    
+    export type HeadersInit = Headers | string[][] | { [key: string]: string | string[] };
+    // HeaderInit is exported to support backwards compatibility. See PR #34382
+    export type HeaderInit = HeadersInit;
+    export type BodyInit =
+        | ArrayBuffer
+        | ArrayBufferView
+        | NodeJS.ReadableStream
+        | string
+        | URLSearchParams
+        | FormData;
+    export type RequestInfo = string | URLLike | Request;
+
+    export function isRedirect(code: number): boolean;
+
+    const _default: typeof fetch;
+
+    export { _default as default };
```

Comment:
@andrewbranch I was hoping to get your general opinion on this refactor + was also surprised that I needed to do the `const _default: typeof X; export { _default as default }` dance in order to satisfy packages importing this with esModuleInterop set to false (hoping to get a better understanding of why) 🙏

Reply:
Never mind on the `esModuleInterop: true` thing. I see it's called out [here](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/MissingExportEquals.md#consequences).